### ISSUE TITLE
Github sign in button on MDN is too tall

### ISSUE THREAD

**Issue by jdm** [servo/servo#14807]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then
Firefox:
<img width="186" alt="screen shot 2016-12-31 at 11 43 00 am" src="https://cloud.githubusercontent.com/assets/27658/21578456/4e04023c-cf4e-11e6-8ea5-602b5a571da1.png">

Servo:
<img width="210" alt="screen shot 2016-12-31 at 11 42 32 am" src="https://cloud.githubusercontent.com/assets/27658/21578452/3d18589c-cf4e-11e6-97ef-f3c9aad5a17d.png">

Very preliminary reduction:
```html
<style>
#nav-sec {
        position: absolute;
        top: 12px;
        right: 200px;
        z-index: 36;
        font-size: 13px;
        font-size: .929rem;
border: 1px solid black;
}
.login-link i[class^="icon-"] {
        float: right;
}
.login i[class^="icon-"] {
        font-size: 15.6px;
        font-size: 1.114rem;
        line-height: 20px;
        vertical-align: top;
}
.icon-github::before {
        content: "\f09b";
}
</style>
<div id="nav-sec">

                  <ul>
                  <li><div class="login">


       <a href="/users/github/login/?next=%2Fen-US%2Fdocs%2FWeb%2FJavaScript%2FReference%2FGlobal_Objects%2FPromise%2Fthen" class="login-link js-login-link" data-service="GitHub">
        Sign in
        <i class="icon-github" aria-hidden="true"></i>
      </a>

</div></li>
                </ul>

    </div>
```

**Comment by DominoTree** [comment]
I'll see if I can fix this one

**Comment by jdm** [comment]
I definitely recommend reducing it to a minimal testcase first.

**Comment by DominoTree** [comment]
Looks like it's the float, the example below still triggers it
```
<style>
.float {
	float: left;
}
</style>
beside this text
<div class="float">This should be</div>
```
Firefox:
![Firefox](https://i.imgur.com/nqa8onR.png)

Servo:
![Servo](https://i.imgur.com/9ZzTfWx.png)

**Comment by DominoTree** [comment]
Think this is #13683

**Comment by stshine** [comment]
Closed in favor of #13683.
